File: JeffSebastian/playground/src/main/java/nl/uva/se/ast/expression/Binary.java
package nl.uva.se.ast.expression;

public abstract class Binary implements Expression {

	private final Expression left;
	private final Expression right;
	
	public Binary(Expression left, Expression right) {
		this.left = left;
		this.right = right;
	}

	public Expression getLeft() {
		return left;
	}

	public Expression getRight() {
		return right;
	}
	
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Binary/Binary.cs
using AST.Nodes.Interfaces;
using AST.Representation;

namespace AST.Nodes.Expressions.Binary
{
    public abstract class Binary : Expression
    {
        private readonly Expression left; 
        private readonly Expression right;

        protected Binary(Expression left, Expression right, PositionInText pos)
            :base(pos)
        {
            this.left = left; 
            this.right = right;
        }

        public Expression Left()
        { return left; }

        public Expression Right()
        { return right; }

        public abstract override T Accept<T>(ASTVisitors.IVisitor<T> visitor);
    }
}
